### PR TITLE
auth part1

### PR DESCRIPTION
**Pull request by Khoacannotcode**
## Related Tickets:
- [#45195](https://edu-redmine.sun-asterisk.vn/issues/45195)

## WHAT this PR does?
- Create Login/Logout/Signup
- Localizaion for Auth

## Notes Impacted Areas:
- N/A.

## Attachments:
- N/A.

### CODE REVIEW

File: app/Http/Controllers/Auth/LoginController.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    public function __construct()
+    {
+        $this->middleware('guest')->except('logout');
+    }
+
+    public function login(Request $request)
+    {
+        $inputVal = $request->all();
+
+        $this->validate($request, [
```
Comment (phuongha47) [inline app/Http/Controllers/Auth/LoginController.php]: phải dùng form request hay sao á

File: app/Http/Controllers/Auth/LoginController.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        $this->middleware('guest')->except('logout');
+    }
+
+    public function login(LoginRequest $request)
+    {
+        $inputVal = $request->all();
+
+        if (auth()->attempt([
+            'email' => $inputVal['email'],
+            'password' => $inputVal['password']
```
Comment (phuongha47) [inline app/Http/Controllers/Auth/LoginController.php]: dấu phẩy cuối

File: app/Models/User.php [hunk 3]
```diff
@@ -4,14 +4,42 @@
...
+class User extends Authenticatable
 {
+    use Notifiable;
+    /**
+     * The attributes that are mass assignable.
+     *
+     * @var array
+     */
+    protected $fillable = [
+        'name', 'email', 'password', 'role_id'
```
Comment (phuongha47) [inline app/Models/User.php]: dấu phẩy

**Comment by phuongha47** [comment]
ready

**Comment by Khoacannotcode** [comment]
ready

File: app/Http/Controllers/Auth/LoginController.php [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    public function login(LoginRequest $request)
+    {
+        $inputVal = $request->all();
+
+        if (auth()->attempt([
+            'email' => $inputVal['email'],
+            'password' => $inputVal['password'],
+            ])) {
```
Comment (ghost) [inline app/Http/Controllers/Auth/LoginController.php]:
```suggestion
        if (auth()->attempt([
            'email' => $inputVal['email'],
            'password' => $inputVal['password'],
        ])) {
```

File: app/Http/Controllers/Auth/LoginController.php [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            if (auth()->user()->role_id == 1) {
+                return redirect()->route('admin.dashboard');
+            } else {
+                return redirect()->route('home');
+            }
+        } else {
+            return redirect()->route('login')
+                ->with('error', 'Email or Password are incorrect.');
```
Comment (ghost) [inline app/Http/Controllers/Auth/LoginController.php]: localization

File: app/Http/Controllers/Auth/LoginController.php [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+    public function login(LoginRequest $request)
+    {
+        $inputVal = $request->all();
+
+        if (auth()->attempt([
+            'email' => $inputVal['email'],
+            'password' => $inputVal['password'],
+            ])) {
+            if (auth()->user()->role_id == 1) {
```
Comment (ghost) [inline app/Http/Controllers/Auth/LoginController.php]: bỏ 1 vào file config

File: app/Http/Middleware/IsAdmin.php [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+     * @param  \Closure(\Illuminate\Http\Request): (\Illuminate\Http\Response|\Illuminate\Http\RedirectResponse)  $next
+     * @return \Illuminate\Http\Response|\Illuminate\Http\RedirectResponse
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        if (auth()->user()->role_id == 1) {
+            return $next($request);
+        }
+        
+        return redirect('home')->with('error', 'Only admin can access!');
```
Comment (ghost) [inline app/Http/Middleware/IsAdmin.php]: localization

File: app/Http/Middleware/IsAdmin.php [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure(\Illuminate\Http\Request): (\Illuminate\Http\Response|\Illuminate\Http\RedirectResponse)  $next
+     * @return \Illuminate\Http\Response|\Illuminate\Http\RedirectResponse
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        if (auth()->user()->role_id == 1) {
```
Comment (ghost) [inline app/Http/Middleware/IsAdmin.php]: bỏ 1 vào config

**Comment by Khoacannotcode** [comment]
ready

**Comment by phuongha47** [comment]
ready

**Review by ghost — APPROVED** [review]

**Pull request merged**